system/jhtree/jhtree.cpp

```diff
@@ -2339,33 +2319,34 @@ extern jhtree_decl void getNodeCacheInfo(ICacheInfoRecorder &cacheInfo)
 void CNodeCache::getCacheInfo(ICacheInfoRecorder &cacheInfo)
 {
     CriticalBlock block(lock);
-    preloadCache.reportEntries(cacheInfo); // Debatable whether we should include this
     blobCache.reportEntries(cacheInfo);
     nodeCache.reportEntries(cacheInfo);
     leafCache.reportEntries(cacheInfo);
 }
 
-CJHTreeNode *CNodeCache::getNode(INodeLoader *keyIndex, unsigned iD, offset_t pos, IContextLogger *ctx, bool isTLK)
+CJHTreeNode *CNodeCache::getNode(INodeLoader *keyIndex, unsigned iD, offset_t pos, NodeType type, IContextLogger *ctx, bool isTLK)
 {
     // MORE - could probably be improved - I think having the cache template separate is not helping us here
     // Also one cache per key would surely be faster, and could still use a global total
     if (!pos)
         return NULL;
-    { 
-        // It's a shame that we don't know the type before we read it. But probably not that big a deal
-        CriticalBlock block(lock);
-        CKeyIdAndPos key(iD, pos);
-        if (preloadNodes)
-        {
-            CJHTreeNode *cacheNode = preloadCache.query(key);
-            if (cacheNode)
-            {
-                cacheHits++;
-                if (ctx) ctx->noteStatistic(StNumPreloadCacheHits, 1);
-                preloadCacheHits++;
-                return LINK(cacheNode);
-            }
-        }
+
+    CKeyIdAndPos key(iD, pos);
+    NodeType cacheType = isTLK ? NodeBranch : type;
+
+    // No benefit in caching the following, especially since they will evict useful pages
+    // could also check cache[cacheType] if an array was used and avoid the tests in the critical section (change later)
+    if ((cacheType == NodeMeta) || (cacheType == NodeBloom))
+    {
+        cacheAdds++; // Not sure if this should be incremented
```

Comment:
I would not have thought so. It's not a cache add (but neither is it a cache miss). Is there a stat for node loads?

Reply:
yes there is a stat for loads - I will remove